src/native/managed/cdac/mscordaccore_universal/Legacy/TargetPointerExtensions.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Diagnostics.DataContractReader.Legacy;
+
+internal static class TargetPointerExtensions
+{
+    /// <summary>
+    /// Converts a TargetPointer to a ClrDataAddress using sign extension if required.
+    /// </summary>
+    public static ClrDataAddress ToClrDataAddress(this TargetPointer address, Target target)
```

Comment:
What happens in the other direction?

DAC checks the address and throws in the other direction: https://github.com/dotnet/runtime/blob/a3f5189c12b92d37840c602f90d9cf786965a5bd/src/coreclr/debug/daccess/dacimpl.h#L56-L68. Do we have the same behavior or do we just ignore the bits?